lib/serialization/booleanSpec.ts

```diff
@@ -18,11 +18,19 @@ const booleanSpec: TypeSpec<boolean, boolean> = {
   },
 
   deserialize(propertyPath: PropertyPath, value: boolean, options: SerializationOptions): boolean {
+    if (typeof value === "string" && options.outputType === SerializationOutputType.XML) {
+      if (value === "true") {
```

Comment:
I think the type of value should be `boolean | string` now. Since having a string is valid in the XML case, it's a bonus perk that we're more flexible in the JSON scenario. That way you also don't have to do the check for whether we're expecting it to be XML or not?

Reply:
@amarzavery and I hashed this out in the old serializer. The thinking is that as much as possible we want to give to the user the values that were produced by the service, except in this case with XML, where the document itself is actually ambiguous (whether the string "true" should be interpreted as string or boolean depends on the TypeSpec)

I'm happy to widen the parameter type though.